visualization/app/codeCharta/ui/fileExtensionBar/fileExtensionBar.component.ts

```diff
@@ -27,10 +36,48 @@ export class FileExtensionBarController implements CodeMapPreRenderServiceSubscr
 		this.potentiallyAddNoneExtension()
 	}
 
+	public highlightBarHoveredBuildings(extension: string) {
+		const mapBuildings: CodeMapBuilding[] = this.threeSceneService
+			.getMapMesh()
+			.getMeshDescription()
+			.buildings.filter(building => building.node.isLeaf)
+		if (extension === "other") {
+			this.findBuildingsSummarizedInOtherDistribution(mapBuildings)
+		} else {
+			mapBuildings.forEach(building => {
+				if (FileExtensionCalculator.estimateFileExtension(building.node.name) === extension) {
+					this.threeSceneService.addBuildingToHighlightingList(building)
+				}
+			})
+		}
+
+		this.threeSceneService.highlightBuildings()
+	}
+
+	private findBuildingsSummarizedInOtherDistribution(mapBuildings: CodeMapBuilding[]) {
```

Comment:
a method that finds something usually returns the found value.
Let's name it `highlightBuildingsSummarizedInOtherDistribution`